### PR TITLE
DHFPROD-5437: Typeahead search for collection source query field in m…

### PR DESCRIPTION
**Pull request by briantang**
…apping step

### Description

- Added typeahead search to Collections field in 'create mapping' and 'create matching' dialogs

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [ ] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/App.scss [hunk 1]
```diff
@@ -1,258 +1,290 @@
 @font-face {
-  font-family: MLCustomFont;
-  src: url(./fonts/ML_IconFont_explore.ttf);
-  font-family: 'MLCustomFont'format('truetype');
-  font-weight: normal;
-  font-style: normal;
+    font-family: MLCustomFont;
```
Comment (rjrudin) [inline marklogic-data-hub-central/ui/src/App.scss:2]: There are a lot of whitespace changes in here - wondering if those will cause conflicts when 5.4-develop is rebased on top of this, or are these whitespace changes already present on 5.4?

Comment (briantang) [reply marklogic-data-hub-central/ui/src/App.scss:2]: I think these were due to tab space discrepancies between IDE's. I can look into reverting those changes.

**Comment by briantang** [comment]
Rebased and fixed formatting spam in .scss files

**Review by wooldridge — CHANGES_REQUESTED** [review]
@briantang, great work! Really nice to have this feature. Unit tests pass.

I'm seeing a minor inconsistency in the Collection field interaction, maybe you can fix it as part of this PR.

When I CREATE a new Mapping step, I see error messaging when the Collection field goes empty. But when I EDIT the settings of that Mapping step and the Collection field goes empty, I don't see the same error messaging appear:

https://drive.google.com/file/d/1OKRFOU1S9nlrClRXFs7apNLXo_lf5iYv/view

Can we make the error messaging appear in both instances?

**Comment by briantang** [comment]
Closing this and reopening due to conflicts.

**Pull request closed without merging**